### PR TITLE
[Velero]: Adding Module

### PR DESCRIPTION
**Pull request by carlosrodlop**
### Testing Notes

```
$> kubectl get all -n velero
NAME                          READY   STATUS    RESTARTS   AGE
pod/velero-6788855749-ghf6h   1/1     Running   0          27m

NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/velero   ClusterIP   172.20.215.88   <none>        8085/TCP   28m

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/velero   1/1     1            1           28m

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/velero-6788855749   1         1         1       28m

$> velero backup-location get
NAME      PROVIDER   BUCKET/PREFIX             PHASE       LAST VALIDATED                  ACCESS MODE   DEFAULT
default   aws        eks-velero-ryan.backups   Available   2022-10-31 10:32:03 +0000 UTC   ReadWrite     true

$> kubectl logs deployment/velero -n velero
...
ller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:114"
time="2022-10-31T10:34:04Z" level=info msg="Backup storage location valid, marking as available" backup-storage-location=default controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.

````

### Configuration

* https://docs.cloudbees.com/docs/admin-resources/latest/backup-restore/velero-dr

### s3 

image

image

### CODE REVIEW

File: modules/aws-velero/main.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  repository = "https://vmware-tanzu.github.io/helm-charts"
+  values     = [local.values]
+  version    = var.chart_version
+}
+
+locals {
+  values = yamlencode({
+    initContainers = [{
+      name            = "velero-plugin-for-aws"
+      image           = "velero/velero-plugin-for-aws:v1.3.0"
```
Comment (ryecarrigan) [inline modules/aws-velero/main.tf:37]: Should image or version be variable to make change easier?

File: modules/aws-velero/variables.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  type = string
+}
+
+variable "k8s_cluster_oidc_arn" {
+  type = string
+}
+
+variable "region_name" {
+  type = string
+}
```
Comment (ryecarrigan) [inline modules/aws-velero/variables.tf:15]: Isn't this support to represent backup region? Could you rename variable to clarify?

**Review by ryecarrigan — CHANGES_REQUESTED** [review]

**Comment by carlosrodlop** [comment]
Thanks @ryecarrigan, I will review all your comments and send a new PR

File: modules/aws-velero/main.tf [hunk 3]
```diff
@@ -0,0 +1,81 @@
+module "velero_eks_role" {
+  source = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+
+  role_name             = "velero-${terraform.workspace}"
```
[the remark this answers is not in the document]
Comment (carlosrodlop) [reply modules/aws-velero/main.tf:4]: updated to `velero-${local.cluster_name}` to know is coming from velero

File: modules/aws-velero/main.tf (hunk 1 above)
Comment (carlosrodlop) [reply modules/aws-velero/main.tf:37]: Good point

File: modules/aws-velero/variables.tf (hunk 2 above)
Comment (carlosrodlop) [reply modules/aws-velero/variables.tf:15]: It has been renamed to `bucket_region` to be more aligned to `bucket_name`

File: roots/eks/main.tf [hunk 4]
```diff
@@ -300,6 +301,58 @@ module "cluster_metrics" {
...
+  }
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (carlosrodlop) [reply roots/eks/main.tf:343]: @ryecarrigan I don't understand this point. Can you explain to me?

**Comment by carlosrodlop** [comment]
Hi @ryecarrigan do you need my help on this?

**Pull request closed without merging**